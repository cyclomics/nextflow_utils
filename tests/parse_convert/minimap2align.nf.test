nextflow_workflow {

    name "Test Mininmap2Align"
    script "parse_convert/subworkflows.nf"
    workflow "Minimap2Align"

    test("Check if consensus mapping to reference with Minimap2 is correct") {

        when {
            workflow {
                """
                sample_id = "SOME_ID_1"
                file_id = "SOME_ID_2"
                fq_file = file("$baseDir/example_datasets/example_native_reads/fastq_filtered/FAW79986_pass_dbcafcd9_40623079_0_filtered.fastq")
                ref = file("$baseDir/example_datasets/reference/pJPVCS_BBCS.mmi")

                input[0] = Channel.of([sample_id, file_id, fq_file])
                input[1] = Channel.of(ref)
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()

            // with(workflow.out.fastq) {
            //    assert path(get(0)[1]).readLines().size() == 11493
            // }
        }
    }
}