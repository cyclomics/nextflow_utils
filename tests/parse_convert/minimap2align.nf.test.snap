{
    "Check if consensus mapping to reference with Minimap2 is correct": {
        "content": [
            {
                "0": [
                    [
                        "SOME_ID_1",
                        "FAW79986_pass_dbcafcd9_40623079_0_filtered.merged",
                        "FAW79986_pass_dbcafcd9_40623079_0_filtered.merged.bam:md5,af60d97b5f741cc1bd5dbd4dcf79186e",
                        "FAW79986_pass_dbcafcd9_40623079_0_filtered.merged.bam.bai:md5,af50a5bddeea3d0d936dec57940770e4"
                    ]
                ],
                "bam": [
                    [
                        "SOME_ID_1",
                        "FAW79986_pass_dbcafcd9_40623079_0_filtered.merged",
                        "FAW79986_pass_dbcafcd9_40623079_0_filtered.merged.bam:md5,af60d97b5f741cc1bd5dbd4dcf79186e",
                        "FAW79986_pass_dbcafcd9_40623079_0_filtered.merged.bam.bai:md5,af50a5bddeea3d0d936dec57940770e4"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-02-05T15:53:16.61924"
    }
}